backend/src/v5/middleware/dataConverter/inputs/teamspaces/projects/models/metadata.js

```diff
@@ -17,37 +17,46 @@
 
 const { createResponseCode, templates } = require('../../../../../../utils/responseCodes');
 const Yup = require('yup');
-const { getMetadataById } = require('../../../../../../models/metadata');
+const { getMetadataByQuery } = require('../../../../../../models/metadata');
 const { respond } = require('../../../../../../utils/responder');
 
 const Metadata = {};
 
 const metadataSchema = Yup.object().shape({
-	key: Yup.string().min(1).max(120).required(),	
+	key: Yup.string().min(1).max(120).required(),
 	value: Yup.lazy((value) => {
 		switch (typeof value) {
-			case 'string':
-				return Yup.string().max(120);
-			case 'number':
-				return Yup.number();
-			case 'boolean':
-				return Yup.bool();			  
-			default:
-				return Yup.mixed().nullable(true).test('ensure-value-present', 'Metadata value is a required field', () => value !== undefined);
-		  }
-	})
+		case 'string':
+			return Yup.string().max(120);
+		case 'number':
+			return Yup.number();
+		case 'boolean':
+			return Yup.bool();
+		default:
+			return Yup.mixed().nullable(true).test('ensure-value-present', 'Metadata value is a required field', () => value !== undefined);
+		}
+	}),
 }).required().noUnknown();
 
-const generateSchema = (existingMetadata) => {
+const generateSchema = (nonCustomMetadataLookup) => {
 	const schema = Yup.object().shape({
-		metadata: Yup.array().of(metadataSchema.test('check-metadata-can-be-Updated', (value, { createError, path }) => {
-			if (existingMetadata.find((m) => m.key === value.key && !m.custom)) {
-				return createError({ path, message: `Metadata ${value.key} already exists and is not editable.` });
+		metadata: Yup.array().of(metadataSchema).required(),
+	}).required().test('check-metadata-can-be-Updated', (value, { createError, path }) => {
+		const nonEditableMetadata = [];
+
+		value.metadata.forEach((metadataToUpdate) => {
```

Comment:
Actually if you just get the keys in as an array (you don't need actually care about the value anyway, it's never used)

Then you can just use [`getCommonElements `]( https://github.com/3drepo/3drepo.io/blob/staging/backend/src/v5/utils/helper/arrays.js#L25) to get `nonEditableMetadata` instead of writing all this